src/draco/javascript/emscripten/decoder_webidl_wrapper.cc

```diff
@@ -91,11 +91,15 @@ long MetadataQuerier::NumEntries(const Metadata &metadata) const {
 const char *MetadataQuerier::GetEntryName(const Metadata &metadata,
                                           int entry_id) {
   if (entry_names_metadata_ != &metadata) {
-    entry_names_.clear();
+    entry_names_.resize(metadata.num_entries());
     entry_names_metadata_ = &metadata;
     // Initialize the list of entry names.
     for (auto &&entry : metadata.entries()) {
-      entry_names_.push_back(entry.first);
+      int name_index = -1;
+      entry.second.GetValue(&name_index);
```

Comment:
wouldn't this assume that the metadata entry is of type "int" .. what if the data has a different type?